[varsha13152] iP

varsha13152 opened the pull request:
# **Tabby: Your To-Do List Companion**

> _"Helping you stay organized, one task at a time."_ 📝

## **Features**
- **Add** tasks to your list.
- **Delete** tasks when done.
- **Find** tasks by keywords.
- **Mark/unmark** tasks as completed.

## **Example Commands**
1. **`todo Buy groceries`** — _Add a new task to your list._  
2. **`delete 2`** — ~~Remove the task you no longer need.~~  
3. **`mark 1`** — **Mark a task as completed.**

## **List of Features**
- [x] **Add tasks**  
- [x] _Delete tasks_  
- [x] **Find tasks**  
- [x] ~~Mark tasks~~  
- [x] **Unmark tasks**  
- [ ] **More coming soon!** 🎉

## **Code Example**
```java
public void addTask(String task) {
    tasks.add(new Task(task));
    System.out.println("Task added: " + task);
}
```

💡 _What are you waiting for?_ **Use it now!!** [Tabby Chatbot](https://github.com/varsha13152/ip)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

varsha13152 committed:
Rename Duke.java to Tabby.java
Update Tabby.java

varsha13152 committed:
Add user input handling

varsha13152 committed:
Created packages tabby and task
Created Task.java
Updated Tabby.java

varsha13152 committed:
Created Class TaskManager, Action, ListAction, AddAction, MarkAction,
UnmarkAction

varsha13152 committed:
Updated Action.java

varsha13152 committed:
Update TaskManager.java

varsha13152 committed:
Update AddAction.java, TaskManager.java
Create Event.java,Deadline.java, ToDo.java

varsha13152 committed:
Revert "Update TaskManager.java"

This reverts commit 91381e51d8fdf85aa1fbcde669bac2f827a0b6db.

varsha13152 committed:
Merge commit 'ad21941641b9ca2b859ff8e3a3ebfa039ab15754'

varsha13152 committed:
Update TaskManager.java

varsha13152 committed:
Update TaskManager.java
Update UI testing files

varsha13152 committed:
Added Error Handling Mechanism

varsha13152 committed:
Create DeleteAction.java
Update Action.java, TaskManager.java,TabbyExceptionInvalidCommand.java

varsha13152 committed:
Update Action.java Update TaskManager.java
Create DeleteAction.java

varsha13152 committed:
Update Action.java,TabbyExceptionInvalidCommand.java

varsha13152 committed:
Update Action.java, TabbyExceptionInvalidCommand.java,
TabbyExceptionInvalidTaskNumber.java
Rename TabbyExceptionInvalidMark.java to TabbyExceptionInvalidInput.java

varsha13152 committed:
Update Action.java, AddAction.java, TabbyExceptionInvalidCommand.java

varsha13152 committed:
Implemented Level-7

varsha13152 committed:
Implement Level-8
Updated Action.java,Parser.java,Tabby.java and
TabbyExceptionIncompleteCommand.java

varsha13152 committed:
Created Storage.java, Updated TaskManager.java

varsha13152 committed:
Create package tabby. Refractor Tabby.java into tabby

varsha13152 committed:
Update Action.java,AddAction.java,Tabby.java,Storage.java
Create Ui.java

varsha13152 committed:
Merge branch 'add-gradle-support'

varsha13152 committed:
Add build.gradle

varsha13152 committed:
Add AddActionTest.java, MarkActionTest.java
Update TaskManager.java

varsha13152 committed:
Add JavaDoc to Action.java,AddAction.java,Ui.java,Storage.java,Task.java
,TaskManager.java

varsha13152 committed:
Update AddAction.java,Tabby.java,TaskManager.java

varsha13152 committed:
Add FindAction.java, Update Action.java, Update TaskManager.java

varsha13152 committed:
Merge branch 'branch-A-CodingStandard'

varsha13152 committed:
Merge branch 'branch-Level-9'

varsha13152 committed:
Update TaskManager.java

varsha13152 committed:
Add checkstyle.xml and suppressions.xml
Update style errors

varsha13152 committed:
Add GUI to application

ziliangstanley commented on src/main/java/action/AddAction.java:
```diff
@@ -0,0 +1,137 @@
+        }
+
+        String taskDescription = input[1];
+        switch (command) {
+            case TODO -> {
+                return addTodoTask(taskManager, taskDescription);
+            }
+            case DEADLINE -> {
+                return addDeadlineTask(taskManager, taskDescription);
+            }
+            case EVENT -> {
+                return addEventTask(taskManager, taskDescription);
```
Should follow coding standards and not indent the case clauses.

ziliangstanley commented on src/main/java/task/TaskManager.java:
```diff
@@ -0,0 +1,184 @@
+     * @param keyword The keyword to search for within task descriptions.
+     * @return A message containing the matching tasks or an error message.
+     */
+    public String findTask(String keyword) {
```
Good job making using verbs to name all the method names in the class.

ziliangstanley commented on src/main/java/tabby/Ui.java:
```diff
@@ -0,0 +1,32 @@
+ * in a formatted and user-friendly manner.
+ */
+public class Ui {
+    private static final String CHATBOT = "Tabby";
```
Good job using SCREAMING_SNAKE_CASE to name this constant.

ziliangstanley commented on src/main/java/action/Action.java:
```diff
@@ -0,0 +1,91 @@
+     * @throws TabbyExceptionInvalidTaskNumber If the task number is not a valid integer.
+     * @throws TabbyExceptionIncompleteCommand If the command is incomplete.
+     */
+    public static Action userAction(String input, boolean isDone, boolean isUserInput, Ui ui)
+            throws TabbyExceptionInvalidCommand, TabbyExceptionInvalidInput,
+            TabbyExceptionInvalidTaskNumber, TabbyExceptionIncompleteCommand {
```
Good compliance with the indentation for a single line spread over a few lines.

ziliangstanley commented on src/main/java/action/AddAction.java:
```diff
@@ -0,0 +1,137 @@
+ */
+public class AddAction extends Action {
+    private final String[] input;
+    private final boolean isDone;
+    private final boolean isUserInput;
```
Good naming for boolean variables

ziliangstanley reviewed (commented):
Very good job. All the best for the rest of the project.

TeaMakesMePee commented on src/main/java/action/Parser.java:
```diff
@@ -0,0 +1,181 @@
+public class Parser {
+
+    // User input patterns
+    private static final Pattern DEADLINE_USER_INPUT_PATTERN = Pattern.compile("(.+?)\\s*/by\\s*(.+)");
+    private static final Pattern EVENT_USER_INPUT_PATTERN = Pattern.compile("(.+?)\\s*/from\\s*(.+?)\\s*/to\\s*(.+)");
+
+    // File input patterns
+    private static final Pattern DEADLINE_FILE_INPUT_PATTERN = Pattern.compile("(.+?)\\s*\\(by:\\s*(.+?)\\)");
+    private static final Pattern EVENT_FILE_INPUT_PATTERN =
+            Pattern.compile("(.+?)\\s*\\(from:\\s*(.+?)\\s*to:\\s*(.+?)\\)");
+    private static final Pattern FILE_INPUT_TASK_PATTERN = Pattern.compile("^\\[(T|E|D)\\]\\[(X| )\\]\\s*(.*)$");
+
+    // Date formatters
+    private static final DateTimeFormatter USER_INPUT_FORMATTER = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
+    private static final DateTimeFormatter OUTPUT_FORMATTER = DateTimeFormatter.ofPattern("MMM dd yyyy, h:mm a");
```
Good naming convention for static final vars!

TeaMakesMePee commented on src/main/java/task/Event.java:
```diff
@@ -0,0 +1,35 @@
+    protected String start;
+    protected String end;
+
+    /**
+     * Constructs a new {@code Event} task with the specified description, start, and end times.
+     *
+     * @param description The description of the event task.
+     * @param start       The start time of the event.
+     * @param end         The end time of the event.
+     */
```
Good job for following the guidelines for code comments!

varsha13152 committed:
Update Action class

The action class parses the user input. In the case the user uses a
single word command, the input is not trimmed.

Trimming the input before submitted the input to Parser class ParseTask
method will mitigate this issue.

Doing so, will prevent an "Invalid Command" error from being thrown

varsha13152 committed:
Update AddAction.java, Parser.java to include assertions

The code previously did not have any assertions to validate that
the user input met the requirements after parsing through the
Parser class.

If the Parser class failed to throw an error, it will lead to run time
errors or undefined behaviors as the parsed input does not meet the
expected format.

Hence, the assertion will prevent the remaining code from executing

varsha13152 committed:
Update files to follow code quality and style standards

Many parts of the codes do not adhere to the code style and quality
standards, making the codebase harder to maintain and read.

Organized, well-structured code improves readability, maintainability,
and ensures long-term consistency.

The codes have been refactored code to enhance readability,
reformat imports, and align with coding quality and style standards.

Adhering to established guidelines promotes future maintainability and
easier collaboration.

varsha13152 committed:
Merge pull request #1 from varsha13152/branch-A-CodeQuality

Branch A-CodeQuality

varsha13152 committed:
Revert "Update files to follow code quality and style standards"

This reverts commit d8f664b152ed60a0329ab3cb65f8da92071bca05.

varsha13152 committed:
Merge pull request #2 from varsha13152/branch-A-Assertions

Revert "Update files to follow code quality and style standards"

varsha13152 committed:
Add new functionality of creating reminders

The application currently lacks a feature to remind users of upcoming
tasks

Without reminders, users may forget important tasks resulting in
missed deadlines

Hence, the chatbot now accepts a new command "reminder" which will
identify Deadline and Event tasks which are marked not done.

This is to identify Deadline and Event tasks that are upcoming but not
completed yet.

varsha13152 committed:
Update GUI in DialogBox.fxml

varsha13152 committed:
Update DialogBox.fxml to add chat bubble GUI

varsha13152 committed:
Update README.md

varsha13152 committed:
Add Ui.png

varsha13152 committed:
Update Action.java

varsha13152 committed:
Revert "Update Action.java"

This reverts commit 5872028f37c2b2cf0b0aeadf03933893b80c1d86.

varsha13152 committed:
Revert "Add Ui.png"

This reverts commit 9485ea02242a799963db3e7523f4a90436090c51.

varsha13152 committed:
Merge branch 'master' of https://github.com/varsha13152/ip

varsha13152 committed:
Update README.md and upload Ui.png

varsha13152 committed:
Update README.md

varsha13152 committed:
Update style issues

varsha13152 committed:
Update Action.java and TaskManager.java to improve code quality

varsha13152 committed:
Update Ui.png filename

varsha13152 committed:
Delete docs/Ui.png.png

Delete Ui.png.png

varsha13152 committed:
Update README.md

varsha13152 committed:
Add javadocs in Action.java

varsha13152 committed:
Update README.md

varsha13152 committed:
Update README.md

varsha13152 committed:
Update README.md

varsha13152 committed:
Update TabbyExceptionIncompleteCommand.java

varsha13152 committed:
Update error messages thrown to be more specific

varsha13152 committed:
Update code styling errors

varsha13152 committed:
Update TabbyExceptionInvalidTaskNumber.java

varsha13152 committed:
Update README.md

varsha13152 committed:
Update README.md

varsha13152 committed:
Update README.md

varsha13152 committed:
Update TabbyExceptionIncompleteCommand.java

varsha13152 committed:
Update IncompleteCommand exception and Invalid Command exception

varsha13152 committed:
Update Parser.java,InvalidEventInput.java, AddActionTest.java

varsha13152 committed:
Update README.md

varsha13152 committed:
Update README.md

varsha13152 committed:
Update README.md

varsha13152 committed:
Update README.md

varsha13152 committed:
Update README.md
